subprojects/core/src/main/java/org/gradle/plugin/repository/rules/PluginModuleOptions.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.plugin.repository.rules;
+
+import org.gradle.api.Incubating;
+import org.gradle.internal.HasInternalProtocol;
+
+/**
+ * Container for the plugin, with options on how to load it.
+ */
+@Incubating
+@HasInternalProtocol
+public interface PluginModuleOptions {
+
+    /**
+     * Will isolate the plugin's classpath from other plugins that are loaded.
+     *
+     * @return this
+     */
+    @Incubating
+    PluginModuleOptions withIsolatedClasspath();
+
+    /**
+     * @return the dependency notation given to {@link PluginDependencyHandler#useModule(java.lang.Object)}
+     */
+    Object getDependencyNotation();
+
+    /**
+     * Changes the name of the plugin that should be applied. This could be used to remove the namespace from
+     * the plugin that was requested.
+     *
+     * @param pluginName Plugin that should be applied from the dependency.
+     * @return this.
+     */
+    PluginModuleOptions withPluginName(String pluginName);
```

Comment:
Example usage: https://github.com/gradle/gradle/pull/883/files#diff-62cdf2f7b5ddb84b021e06e2b670aa44R149

An example would be I have a plugin that I build today called 'mywar' I want to use this plugin with the rules based plugin, but I also want it to be namespaces (like Gradle did with their plugins org.gradle.java). So I tell users to use the id `com.acme.mywar` then the rule will know that the real name of the plugin that should be applied is mywar and it all works.